spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/OptimizerWrapper.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.example.optimizerWrapper
+
+import java.nio.file.Paths
+
+import com.intel.analytics.bigdl.dataset.{DistributedDataSet, MiniBatch}
+import com.intel.analytics.bigdl.dataset.image.{BytesToGreyImg, GreyImgNormalizer, GreyImgToBatch}
+import com.intel.analytics.bigdl.models.lenet.{LeNet5, Utils}
+import com.intel.analytics.bigdl.models.lenet.Utils._
+import com.intel.analytics.bigdl.nn.Module
+import com.intel.analytics.bigdl.utils.LoggerFilter
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.ml.{DLEstimator, Pipeline}
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.dataset.{DataSet, _}
+import com.intel.analytics.bigdl.example.imageclassification.MlUtils.{testMean => _, testStd => _, _}
+import com.intel.analytics.bigdl.nn.ClassNLLCriterion
+import com.intel.analytics.bigdl.utils.{Engine, T}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.mllib.linalg.DenseVector
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * An example to show how to use DLEstimator fit to be compatible with ML Pipeline
+ * refer to README.md on how to run this example
+ */
+object OptimizerWrapper {
```

Comment:
How about DLEstimatorExample? or if you have plan to add other example for DLEstimator, then we can name it DLEstimatorLeNetExample.
We need to avoid confusion here and try to be specific.